### PR TITLE
Add tweedle curves

### LINKED ISSUE #4: Add TweedleDee/Dum
These curves are implemented in the [O(1)-labs fork](https://github.com/o1-labs/zexe/tree/master/algebra/src/tweedle), so porting them over is a simple copy-paste job.

### PR DESCRIPTION
**Pull request by therealyingtong**
Closes #4 

Adapted from https://github.com/o1-labs/zexe/tree/master/algebra/src/tweedle

### CODE REVIEW

**Comment by Pratyush** [comment]
Hey @therealyingtong !

Thanks for the awesome PR! I'll try to get a review to you ASAP!

File: tweedle/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+[package]
+name = "ark-tweedle"
+version = "0.1.0"
+authors = [ "arkworks contributors" ]
+description = "The Tweedledum/Tweedledee curve cycle"
+homepage = "https://arkworks.rs"
+repository = "https://github.com/arkworks/curves"
+documentation = "https://github.com/daira/tweedle/"
```
Comment (Pratyush) [inline tweedle/Cargo.toml:8]:
```suggestion
documentation = "https://docs.rs/ark-tweedle/"
```

File: tweedle/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+[package]
+name = "ark-tweedle"
+version = "0.1.0"
+authors = [ "arkworks contributors" ]
```
Comment (Pratyush) [inline tweedle/Cargo.toml:4]:
```suggestion
authors = [ "arkworks contributors", "O(1) Labs" ]
```

**Comment by Pratyush** [comment]
Couple of quick comments:
* Do you mind splitting the curves into separate crates, like how it's done for the `MNT` cycles?
* It would be great if you could also add constraints for these, again following the pattern in the `MNT` crates. Note that you wouldn't have to handle anything related to pairings, as the Tweedle curves aren't pairing-friendly. If you'd rather handle that separately, then we do it in a separate PR too =)

**Comment by therealyingtong** [comment]
thanks for the comments! i've split them into `tweedledee` and `tweedledum` modules.

i got a `SerializationError::NotEnoughSpace` when trying to run `sw_tests::<TweedledumParameters>()` or `sw_tests::<TweedledumParameters>()`; but the test uses a buffer size of 32 bytes which seems correct. would you have any guesses re: the cause of this error?

File: README.md [hunk 3]
```diff
@@ -29,3 +29,6 @@ This repository contains implementations of some popular elliptic curves. The cu
 * [`ark-mnt4-753`](mnt4-753): Implements the MNT4-753 pairing-friendly curve. This curve forms a pairing-friendly cycle with MNT6-753
 * [`ark-mnt6-753`](mnt6-753): Implements the MNT6-753 pairing-friendly curve. This curve forms a pairing-friendly cycle with MNT4-753
 * [`ark-ed-on-mnt4-753`](ed-on-mnt4-753): Implements a Twisted Edwards curve atop the scalar field of MNT4-753
+
+### Tweedle cycle of curves
+* [`ark-tweedle`](tweedle): Implements Tweedledum and Tweedledee, an amicable pair of prime-order curves
```
Comment (Pratyush) [inline README.md:34]:
```suggestion
### Tweedle cycle of curves
* [`ark-tweedledee`](tweedledee): Implements Tweedledee, a prime order curve that forms an amicable pair with Tweedledum.
* [`ark-tweedledum`](tweedledum): Implements Tweedledum, a prime order curve that forms an amicable pair with Tweedledee.
```

**Comment by daira** [comment]
At ECC we have switched entirely to [Pallas/Vesta](https://electriccoin.co/blog/the-pasta-curves-for-halo-2-and-beyond/) for use with Halo 2, and we don't intend to deploy the Tweedledum/Tweedledee curves or promote their adoption. Pallas/Vesta are implemented in #21, so I think this can be closed.

**Pull request closed without merging**